narayana/ArjunaJTA/jta/tests/classes/io/narayana/perf/product/BitronixWorker.java

```diff
@@ -21,114 +21,29 @@
-import bitronix.tm.resource.jdbc.PoolingDataSource;
-import bitronix.tm.resource.jdbc.lrc.LrcXADataSource;
-import com.arjuna.ats.jta.xa.performance.XAResourceImpl;
-import io.narayana.perf.product.btm.BtmXAResourceHolderState;
 import io.narayana.perf.product.btm.BtmResourceBean;
+import io.narayana.perf.product.btm.BtmXAResourceHolderState;
 import io.narayana.perf.product.btm.BtmXAResourceProducer;
-
-import javax.transaction.TransactionManager;
-import javax.transaction.xa.XAResource;
-import java.util.Properties;
-import java.util.concurrent.atomic.AtomicInteger;
+import bitronix.tm.recovery.RecoveryException;
+import bitronix.tm.resource.ResourceRegistrar;
+import bitronix.tm.resource.common.XAResourceProducer;
 
 public class BitronixWorker<Void> extends ProductWorker<Void> {
 
+    private XAResourceProducer xaResourceProducer;
+
     public BitronixWorker(ProductInterface prod) {
         super(prod);
     }
-
-    private TransactionManager getTM() {
-        return prod.getTransactionManager();
-    }
-
-    private void initProps(PoolingDataSource pds, Properties properties) {
-        properties.put("user", "sa");
-        properties.put("password", "sa");
-        properties.put("url", "jdbc:h2:tcp://localhost/~/jbpm-db;MVCC=TRUE");
-        properties.put("driverClassName", "org.h2.Driver");
-    }
-
-    private void xinitProps(PoolingDataSource pds, Properties properties) {
-        pds.getDriverProperties().put("user",properties.getProperty("persistence.datasource.user","sa"));
-        pds.getDriverProperties().put("password",properties.getProperty("persistence.datasource.password","sa"));
-        pds.getDriverProperties().put("url",properties.getProperty("persistence.datasource.url","jdbc:h2:tcp://localhost/~/jbpm-db;MVCC=TRUE"));
-        pds.getDriverProperties().put("driverClassName",properties.getProperty("persistence.datasource.driverClassName","org.h2.Driver"));
-    }
-
-/*    public void testRecycleAfterSuspend() throws Exception {
-        PoolingDataSource pds = new PoolingDataSource();
-
-        pds.setClassName(LrcXADataSource.class.getName());
-        pds.setUniqueName("lrc-pds");
-        pds.setMaxPoolSize(2);
-        initProps(pds, pds.getDriverProperties());
-//        pds.getDriverProperties().setProperty("driverClassName", org.h2.jdbcx.JdbcDataSource.class.getName());
-        pds.init();
-
-        getTM().begin();
-
-        Connection c1 = pds.getConnection();
-        c1.createStatement();
-        c1.close();
-
-        Transaction tx = getTM().suspend();
-
-        getTM().begin();
-
-        Connection c11 = pds.getConnection();
-        c11.createStatement();
-        c11.close();
-
-        getTM().commit();
-
-
-        getTM().resume(tx);
-
-        Connection c2 = pds.getConnection();
-        c2.createStatement();
-        c2.close();
-
-    }*/
-
-    @Override
-    public XAResource getXAResource() {
-        return xars[xarCounter.getAndIncrement() % xars.length];
-    }
```

Comment:
This is where the collateral damage occurred. If you really want to get rid of the call then make sure the bitronix worker gets the XAResources from the super class